apps/src/dance/Dance.js

```diff
@@ -287,6 +287,13 @@ Dance.prototype.afterInject_ = function () {
   }
 };
 
+Dance.prototype.playSong = function (url, callback, onEnded) {
+  audioCommands.playSound({url: url, callback: callback, onEnded: () => {
+    onEnded();
+    this.studioApp_.toggleRunReset('run');
```

Comment:
My first question was whether we should be calling toggleRunReset in onEnded itself.

It looks like that function is created here https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/dance/Dance.js#L428 and just resolves a promise.

Execute is called here https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/dance/Dance.js#L385 and then immediately calls `toggleRunReset('reset')`. Should it instead be calling `toggleRunReset('run')` immediately followed by the reset?

(also very possible im misunderstanding something)

Reply:
I'm not quite following your suggestion - On Dance.js line 428, we are creating the callback to be used here: https://github.com/code-dot-org/dance-party/blob/1f4d1d7fc0e6c407e451f060f25795d58c6415e7/src/p5.dance.js#L232. So we do want Dance.js Line 385 to execute the song (to trigger the play) and then immediately call `toggleRunReset('reset')` so that when the song starts to play, we set the run button to be reset.

Is your suggestion to pass the toggleRunReset('run') into onEnded() function (like onEnded(() => {toggleRunReset('run')}) that then gets executed from inside p5.dance?

Reply:
Sorry, I think I was confusing the function passed in as a callback and the function passed in for `onEnded`, leading me to say things that didn't make sense.